c10/util/Half.h

```diff
@@ -51,6 +51,16 @@
 #include <arm_neon.h>
 #endif
 
+#if defined(__GNUC__) || defined(__clang__)
+#if defined(__x86_64__) || defined(_M_X64) || defined(__i386) || \
+    defined(_M_IX86)
+#if defined(__F16C__)
+#define X86_F16 1
```

Comment:
Should it be something like `_HAS_X86_64_F16` (to avoid even a slight possibility of name collision)